src/snowflake/cli/plugins/snowpark/package_utils.py

```diff
@@ -68,165 +66,183 @@ def deduplicate_and_sort_reqs(
     return deduped
 
 
-def install_packages(
-    anaconda: AnacondaChannel,
+def get_snowflake_packages() -> List[str]:
+    requirements_file = SecurePath("requirements.snowflake.txt")
+    if requirements_file.exists():
+        with requirements_file.open(
+            "r", read_file_limit_mb=DEFAULT_SIZE_LIMIT_MB, encoding="utf-8"
+        ) as f:
+            return [req for line in f if (req := line.split("#")[0].strip())]
+    else:
+        return []
+
+
+def generate_deploy_stage_name(identifier: str) -> str:
+    return (
+        identifier.replace("()", "")
+        .replace(
+            "(",
+            "_",
+        )
+        .replace(
+            ")",
+            "",
+        )
+        .replace(
+            " ",
+            "_",
+        )
+        .replace(
+            ",",
+            "",
+        )
+    )
+
+
+def download_packages(
+    anaconda: AnacondaChannel | None,
     requirements_file: SecurePath | None,
     packages_dir: SecurePath,
-    perform_anaconda_check: bool = True,
+    perform_anaconda_check_for_dependencies: bool = True,
     package_name: str | None = None,
-    allow_native_libraries: PypiOption = PypiOption.ASK,
+    index_url: str | None = None,
+    allow_shared_libraries: PypiOption = PypiOption.ASK,
+    skip_version_check: bool = False,
 ) -> tuple[bool, SplitRequirements | None]:
     """
-    Install packages from a requirements.txt file or a single package name,
-    into a local folder named '.packages'.
-    If a requirements.txt file is provided, they will be installed using pip.
-    If a package name is provided, it will be installed using pip.
+    Downloads packages from a requirements.txt file or a single package name,
+    into a local directory named '.packages'.
+    If perform_anaconda_check_for_dependencies is set to True, dependencies available in Snowflake Anaconda
+    channel will be omitted, otherwise all packages will be downloaded using pip.
+
     Returns a tuple of:
-    1) a boolean indicating whether the installation was successful
+    1) a boolean indicating whether the download was successful
     2) a SplitRequirements object containing any installed dependencies
-    which are available on the Snowflake Anaconda channel. These will have
-    been deleted from the local packages folder.
+        which are available on the Snowflake Anaconda channel.
+        They will not be downloaded into '.packages' directory.
     """
     if requirements_file and package_name:
         raise ClickException(
             "Could not use package name and requirements file simultaneously"
         )
-
     if requirements_file and not requirements_file.exists():
         raise ClickException(f"File {requirements_file.path} does not exists.")
+    if perform_anaconda_check_for_dependencies and not anaconda:
+        raise ClickException(
+            "Cannot perform anaconda checks if anaconda channel is not specified."
+        )
```

Comment:
How users can specify Anaconda channel in the command? This message is implementation detail.

One solution to that is to make AnacondaChannel a singleton so we can cache dependencies and easily access them without having to push instance between commands. WDYt?

Reply:
If it is some kind of "illegal state exception", then IMO it shouldn't be ClickException which is designed to be used to inform users about errors.

Reply:
+1 to what @sfc-gh-pjob says, to me it sounds like `RuntimeError`

Reply:
This case is removed in
https://github.com/snowflakedb/snowflake-cli/pull/937, but I get the point :+1: